ci/scripts/run-ecs-db-migration.js

```diff
@@ -90,7 +99,13 @@ const run = async function run () {
     }
 
     console.log(`Succesfully scheduled db migration task: ${runResult.tasks[0].taskArn}`)
-    console.log('View progress of the migration at: https://gds.splunkcloud.com/en-GB/app/gds-004-pay/migration_status')
+
+    await waitFor({
+      cluster: CLUSTER_NAME,
+      tasks: [`${runResult.tasks[0].taskArn}`]
+    })
+
```

Comment:
I think after the task is stopped you should query what the exit code of the task is, if it's a failure state exit out of this with a failure. Tasks that fail also go into the STOPPED state so a failed migration won't throw any exception.

Reply:
What about failing the job if the status of the task is anything other than "RUNNING"?

Reply:
The available states are at https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-lifecycle.html

Reply:
So something like 
```
    const waitForResult = await waitFor({
      cluster: CLUSTER_NAME,
      tasks: [`${runResult.tasks[0].taskArn}`]
    })

    if (waitForResult.tasks[0].lastStatus !== 'RUNNING') {
      console.log(`Expected task status 'RUNNING' but was ${waitForResult.tasks[0].lastStatus}.`)
      process.exit(1)
    }
```

Reply:
The state immediately prior to STOPPED is very likely to be DEPROVISIONING rather than RUNNING though. I think you need to use the describe tasks api (I'm sure there's an aws-sdk equivalent, or possibly it's in the waitForResult too) https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTasks.html to the the exit code (https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Container.html#ECS-Type-Container-exitCode) for the specific container (I think we actually have multiple containers running in the task (e.g. telegraf, nginx, the app, in some cases an nginx-forward-proxy (maybe only in frontend for that one)) and check it's exitCode is 0.

Reply:
https://github.com/alphagov/pay-ci/pull/685/commits/50da474341146b68d67f1d5240b02cc1927ced48.
[Successful DB migration](https://pay-cd.deploy.payments.service.gov.uk/teams/pay-dev/pipelines/deploy-to-test/jobs/adminusers-db-migration/builds/10).

Reply:
Hmm yeah I think that'll be super painful to test. Because a poorly formatted migration script for example would fail the app integration test to begin with.